loader/src/reference_binder.py

```diff
@@ -0,0 +1,166 @@
+import re
+from collections import defaultdict
+from dotty_dict import dotty
+
+from loader.src.config.logger import create_logger
+
+logger = create_logger("reference_binder")
+
+
+# dotty-dict does not handle brackets indices,
+# it uses dots instead (a.0.b instead of a[0].b)
+def dotty_paths(paths):
+    for path in paths:
+        yield re.sub(r"\[(\d+)\]", r".\1", path)
+
+
+def partial_identifier(identifier):
+    (
+        value,
+        system,
+        identifier_type_code,
+        identifier_type_system,
+    ) = ReferenceBinder.extract_key_tuple(identifier)
+    if value:
```

Comment:
If having both value/system and type would cause errors, can you add something to log why they happened?

Reply:
updated the extract_key_tuple to do that